source/MaterialXRuntime/Commands/MetadataCommands.h

```diff
@@ -0,0 +1,29 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTMETADATACOMMANDS_H
+#define MATERIALX_RTMETADATACOMMANDS_H
+
+/// @file
+/// Commands for connection handling.
+
+#include <MaterialXRuntime/RtCommand.h>
+#include <MaterialXRuntime/RtToken.h>
+
+namespace MaterialX
+{
+
+namespace RtCommand
+{
+    /// Set a metadata value from data given as a value string. Creates the metadata if it doesn't exist.
+    void setMetadataFromString(RtPrim& src, const RtToken& metadataName, const string& metadataValue, RtCommandResult& result);
```

Comment:
'src' could be an RtObject instead of RtPrim, since metadata can be set on both prims and inputs/outputs.

Reply:
Could pass these as const RtObject&, since it's not an in/out argument.